client/payments/data/test/reducer.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @jest-environment node
+ */
+
+/**
+ * Internal dependencies
+ */
+import reducer from '../reducer';
+import { ACTION_TYPES as TYPES } from '../action-types';
+import { PluginsState } from '../types';
+import { paymentGatewaysStub } from './stub';
+
+const defaultState: PluginsState = {
+	paymentGateways: [],
+	requesting: {},
+	errors: {},
+};
+
+describe( 'plugins reducer', () => {
+	it( 'should return a default state', () => {
+		const state = reducer( undefined );
+		expect( state ).toEqual( defaultState );
+		expect( state ).not.toBe( defaultState );
+	} );
+
+	it( 'should handle GET_PAYMENT_GATEWAYS_SUCESS', () => {
+		const state = reducer( defaultState, {
+			type: TYPES.GET_PAYMENT_GATEWAYS_SUCESS,
+			paymentGateways: paymentGatewaysStub,
+		} );
+
+		expect( state.paymentGateways ).toHaveLength( 2 );
+		expect( state.paymentGateways ).toBe( paymentGatewaysStub );
+	} );
+
+	it( 'should replace an existing payment gateway on UPDATE_PAYMENT_GATEWAY_SUCESS', () => {
+		const updatedPaymentGateway = {
+			...paymentGatewaysStub[ 1 ],
+			description: 'update test',
+		};
+		const state = reducer(
+			{
+				...defaultState,
+				paymentGateways: paymentGatewaysStub,
+			},
+			{
+				type: TYPES.UPDATE_PAYMENT_GATEWAY_SUCESS,
```

Comment:
It appears this isn't part of the `TYPES` yet, also you missed a `C` in success :)